server/src/main/java/org/opensearch/index/translog/BufferedChecksumStreamInput.java

```diff
@@ -117,7 +118,11 @@ public void reset() throws IOException {
 
     @Override
     public int read() throws IOException {
-        return readByte() & 0xFF;
+        try {
+            return readByte() & 0xFF;
+        } catch (EOFException e) {
+            return -1;
+        }
```

Comment:
why do we need this change ?

Reply:
This is due to an interface contract that OpenSearch was breaking. This is unrelated to the change

Reply:
It looks like we have a [base contract test](https://github.com/opensearch-project/OpenSearch/blob/25955782b900454f40e4baa0688d5869162808b0/server/src/test/java/org/opensearch/common/io/stream/BaseStreamTests.java) for a stream input. Can you create an issue to create a contract test for this class since you've got the full context for this specific miss?